### PR TITLE
Add support for LSP code action grouping

### PR DESCRIPTION
**Pull request by allisonchou**
Adds grouping for LSP code action sets.

Before grouping:
beforegrouping

After grouping:
aftergrouping

### CODE REVIEW

File: src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActionsSource.cs
Comment (allisonchou) [inline src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActionsSource.cs:345]: This was a bug in the existing code that used the end point instead of the length.

File: src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs
Comment (allisonchou) [inline src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:106]: Currently grouping is done by prefixing "Roslyn" with the group number. "Roslyn" was prefixed just to make sure no other language servers use the same group name. If anyone has better suggestions for group names, please let me know.

File: src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActionsSource.cs
Comment (allisonchou) [inline src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActionsSource.cs:287]: Explicitly changed the priority of "Suppress or Configure issues" to low so that it will always show at the bottom of the code actions list, matching the local experience.

File: src/Features/LanguageServer/ProtocolUnitTests/CodeActions/CodeActionsTests.cs
Comment (allisonchou) [inline src/Features/LanguageServer/ProtocolUnitTests/CodeActions/CodeActionsTests.cs:45]: Specifying "Roslyn1" seems to work for now, but may be a little risky in the future if the first code action in the list changes to something else in the future.

File: src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs
Comment (davidwengier) [inline src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:120]: Nit: Could change to `is not` and early exit.

**Review by davidwengier — APPROVED** [review]
LGTM. This will go a long way to making live share feel more "real" in my opinion.

File: src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs
Comment (allisonchou) [reply src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:250]: If Roslyn gives no priority to the code action, then I think LSP should decide the priority. I'll add a comment to indicate that though, thanks!

File: src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActionsSource.cs
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActionsSource.cs:287]: shouldn't this be copying an existing priority over?  i.e. instead of hardcoding this in here, we shoudl grab the priority of the actual code action that the other parts of our system is generating?

File: src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActionsSource.cs
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActionsSource.cs:345]:
```suggestion
                var combinedSpan = minStart >= 0 ? TextSpan.FromBounds(minStart, maxEnd) : (TextSpan?)null;
```

File: src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs
Comment (CyrusNajmabadi) [reply src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:106]: this seems reasonable.

File: src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:106]:
```suggestion
                Group = "Roslyn" + currentSetNumber,
```

or

```suggestion
                Group = $"Roslyn{currentSetNumber}",
```

File: src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:50]:
```suggestion
            var currentGroupId = 0;
```

File: src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:252]: this doesn't seem reacahble.  throw if you hit this.

Comment (allisonchou) [reply src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:252]: There's a [CodeActionPriority.None](http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/CodeActions/CodeActionPriority.cs,14), in which case I thought it would make the most sense to let LSP decide the priority. I can make it an explicit case for clarity if you think that would be best.

Comment (davidwengier) [reply src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:252]: An explicit mapping of `None` to `null`, and a `_ => throw Unreachable` makes sense to me

File: src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:252]: so, tbh, i'm surprised we have a 'none' priority.  i think it's fine for you to do this.  but my actua preference is that we just outright remove this value from roslyn itself.  i see three usages of this, and i don't think any of them should actually exist (they should just not set any value, or set their value to 'medium').

Comment (allisonchou) [reply src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:252]: I think LSP will need to add a 4th enum value. Will check them whether they are willing to do this.

In the meantime, is this PR considered blocked until then? I've temporarily assigned `Suppress or Configure issues` to `CodeActionPriority.Low`. Is this OK for the meantime, if we add a tracking bug for the 4th enum value to the code?

Comment (allisonchou) [reply src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs]: Update, LSP has added a 4th enum type (thanks Miguel!) so I will wait to merge this PR until that change goes in: https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient/pullrequest/280778

**Pull request merged**
